src/core/main.c

```diff
@@ -1590,6 +1582,16 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
+        /* Initialize default unit */
+        if (!arg_default_unit) {
+                r = free_and_strdup(&arg_default_unit, SPECIAL_DEFAULT_TARGET);
```

Comment:
@poettering Yes right, what if we replace the `free_and_strdup` with `strdup` also in the `parse_proc_cmdline_item` for `arg_default_unit` as it first place now where `arg_default_unit` will be touched and it is NULL everytime there.